lib/pbench/test/unit/server/database/test_index_map_db.py

```diff
@@ -37,69 +37,91 @@ def test_create(self, db_session, attach_dataset):
+                "run-sample": {"prefix.run-sample.2023-08": ["id22", "id23"]},
             },
-            "run-sample": {"prefix.run-sample.2023-08": ["id22", "id23"]},
-        }
-        drb = Dataset.query(name="drb")
-        IndexMap.create(drb, map1)
-        IndexMap.merge(drb, map2)
-
-        map = list(IndexMap.stream(drb))
-        ids = [m.id for m in map]
-        indices = set(m.index for m in map)
-        assert sorted(indices) == [
-            "prefix.run-data.2023-06",
-            "prefix.run-data.2023-07",
-            "prefix.run-sample.2023-08",
-            "prefix.run-toc.2023-06",
-            "prefix.run-toc.2023-07",
-            "prefix.run-toc.2023-08",
         ]
-        assert sorted(ids) == [
-            "id1",
-            "id10",
-            "id12",
-            "id13",
-            "id14",
-            "id2",
-            "id20",
-            "id21",
-            "id22",
-            "id23",
-            "id3",
-            "id4",
-            "id5",
-            "id6",
-            "id8",
-            "id9",
+        dataset = Dataset.query(name="drb")
+        assert not IndexMap.exists(dataset)
+        assert list(IndexMap.stream(dataset)) == []
+
+        IndexMap.create(dataset, maps[m1])
+        IndexMap.merge(dataset, maps[m2])
+
+        assert sorted(IndexMap.stream(dataset), key=lambda i: (i.id, i.index)) == [
+            IndexStream("prefix.funky-data.1918-02", "id1"),
+            IndexStream("prefix.run-data.2023-07", "id1"),
+            IndexStream("prefix.run-toc.2023-06", "id10"),
+            IndexStream("prefix.run-toc.2023-06", "id12"),
+            IndexStream("prefix.run-toc.2023-06", "id13"),
+            IndexStream("prefix.run-toc.2023-06", "id14"),
+            IndexStream("prefix.funky-data.1918-02", "id2"),
+            IndexStream("prefix.run-data.2023-07", "id2"),
+            IndexStream("prefix.run-toc.2023-08", "id20"),
+            IndexStream("prefix.run-toc.2023-08", "id21"),
+            IndexStream("prefix.run-sample.2023-08", "id22"),
+            IndexStream("prefix.run-sample.2023-08", "id23"),
+            IndexStream("prefix.run-toc.2023-06", "id3"),
+            IndexStream("prefix.run-toc.2023-06", "id4"),
+            IndexStream("prefix.run-toc.2023-07", "id5"),
+            IndexStream("prefix.run-toc.2023-07", "id6"),
+            IndexStream("prefix.run-data.2023-06", "id8"),
+            IndexStream("prefix.run-data.2023-06", "id9"),
         ]
 
-    def test_merge_none(self, db_session, attach_dataset):
-        """Test index map merge with no existing map
+    @pytest.mark.parametrize("to", (False, True))
+    def test_merge_none(self, db_session, attach_dataset, to):
+        """Test index map merge with an empty map
 
-        The result should be the new map.
+        We test two cases: one merging a map into an empty map, and then
+        merging an empty map into an existing map.
+
+        In either case the result should be the single map.
```

Comment:
You didn't want to test the third case -- merging a (non-empty) map into itself?  You could do that using this test substantially as it is (i.e., it would do the same assertion after similar setup), you just need to parameterize the source and merge maps separately.

Reply:
Yeah, I remembered that while I was hiking, as often happens.

Reply:
Actually, `merge` wasn't designed to de-dup document IDs, so merging `map` into `map` doubles everything. Since the intended use case doesn't need that feature, I'm going to leave it alone.